File: internal/domain/entity/race.go
package entity

import "time"

type Race struct {
	ID        int64     `json:"id"         db:"id"`
	ClubID    int64     `json:"club_id"    db:"club_id"`
	Date      time.Time `json:"date"       db:"date"`
	Type      string    `json:"type"       db:"type"`
	Place     string    `json:"place"      db:"place"`
	Distances string    `json:"distances"  db:"distances"`
	Name      string    `json:"name"       db:"name"`
	CreatedAt time.Time `json:"created_at" db:"created_at"`
	UpdatedAt time.Time `json:"updated_at" db:"updated_at"`
}

type RaceRegistration struct {
	ID        int64     `json:"id"         db:"id"`
	RaceID    int64     `json:"race_id"    db:"race_id"`
	MemberID  int64     `json:"member_id"  db:"member_id"`
	Distance  string    `json:"distance"   db:"distance"`
	CreatedAt time.Time `json:"created_at" db:"created_at"`
}

File: web/admin/src/api/race.ts
import client from './client'

export interface Race {
  id: number
  club_id: number
  name: string
  date: string
  type: string
  place: string
  distances: string
  created_at?: string
  updated_at?: string
}

export interface RaceCreate {
  name: string
  date: string
  type: string
  place: string
  distances: string
}

export interface RaceUpdate {
  name?: string
  date?: string
  type?: string
  place?: string
  distances?: string
}

export interface Registration {
  id: number
  race_id: number
  member_id: number
  distance: string
  created_at?: string
}

export interface RegistrationCreate {
  member_id: number
  distance: string
}

export default {
  list(clubId: number) {
    return client.get<Race[]>(`/clubs/${clubId}/races`)
  },

  create(clubId: number, data: RaceCreate) {
    return client.post<Race>(`/clubs/${clubId}/races`, data)
  },

  get(id: number) {
    return client.get<Race>(`/races/${id}`)
  },

  update(id: number, data: RaceUpdate) {
    return client.put<Race>(`/races/${id}`, data)
  },

  remove(id: number) {
    return client.delete(`/races/${id}`)
  },

  registrations(raceId: number) {
    return client.get<Registration[]>(`/races/${raceId}/registrations`)
  },

  register(raceId: number, data: RegistrationCreate) {
    return client.post(`/races/${raceId}/registrations`, data)
  },

  unregister(raceId: number, data: { member_id: number }) {
    return client.delete(`/races/${raceId}/registrations`, { data })
  },
}
